### PR TITLE
Update pattern matching tutorial to .NET 6

### PR DESCRIPTION
**Pull request by stvansolano**
## Summary

Contributes to #4953 For 'Pattern matching tutorial' + some C# 9 tweaks

Also keep the files consistent with the tutorial samples from [docs/csharp/fundamentals/tutorials/pattern-matching.md](https://github.com/dotnet/docs/blob/main/docs/csharp/fundamentals/tutorials/pattern-matching.md) aka live -> https://docs.microsoft.com/dotnet/csharp/fundamentals/tutorials/pattern-matching

cc @adegeo @BillWagner @IEvangelist

### CODE REVIEW

File: csharp/tutorials/patterns/start/toll-calculator/Program.cs [hunk 1]
```diff
@@ -3,126 +3,119 @@
 using ConsumerVehicleRegistration;
 using LiveryRegistration;
 
-namespace toll_calculator
+using toll_calculator;
```
Comment (BillWagner) [inline csharp/tutorials/patterns/start/toll-calculator/Program.cs:6]: This looks like a copy / paste error:

```suggestion
namespace toll_calculator;
```

**Review by BillWagner — CHANGES_REQUESTED** [review]
This is great work @stvansolano Thank you!

I noticed one mistake that needs to be addressed before we merge this.

**Comment by stvansolano** [comment]
Thanks @BillWagner looks good to me. Feel free to commit any other change you consider appropriate.

**Comment by stvansolano** [comment]
@BillWagner I added a new change in order to prevent build errors on `/start` Program.cs file given that was supposed to be empty. So keeping some lines on code out of comments makes the build and top-level program happy.

**Review by BillWagner — APPROVED** [review]

**Comment by BillWagner** [comment]
Thanks for fixing the build issues @stvansolano 

I'll :shipit: now.

**Pull request merged**